cmd/k8s.go

```diff
@@ -40,6 +40,26 @@ func (k8s *k8s) watchForPods(podManager kcache.ResourceEventHandler) kcache.Stor
 	return podStore
 }
 
+// returns a listwatcher of namespaces
+func (k8s *k8s) createNamespaceLW() *kcache.ListWatch {
+	return kcache.NewListWatchFromClient(k8s, "namespaces", api.NamespaceAll, selector.Everything())
+}
+
+func (k8s *k8s) watchForNamespaces(nsManager kcache.ResourceEventHandler) kcache.Store {
+	nsStore, nsController := kcache.NewInformer(
+		k8s.createNamespaceLW(),
+		&api.Namespace{},
+		resyncPeriod,
+		kcache.ResourceEventHandlerFuncs{
+			AddFunc:    nsManager.OnAdd,
+			DeleteFunc: nsManager.OnDelete,
+			UpdateFunc: nsManager.OnUpdate,
+		},
```

Comment:
Just pass `nsManager` directly, no need to wrap it in another layer of indirection.